Report reference error with sandpack-react/unstyled in typescript.

# Bug report

## Packages affected

-   [ ] sandpack-client
-   [x] sandpack-react

## Description of the problem

<img width="1448" alt="image" src="https://user-images.githubusercontent.com/12525415/225542385-dd16f09f-d711-4c6e-a73d-8cba12074598.png">

### What steps can we take to reproduce the problem?

- create react-ts with vite 

`npm create vite@latest react-app -- --template react-ts`

- edit App.tsx

```js
import { Sandpack } from '@codesandbox/sandpack-react';
import { Sandpack as SandpackUnstyled } from '@codesandbox/sandpack-react/unstyled';

function App() {
  return (
    <div className="App">
      <Sandpack />
      <br />
      <SandpackUnstyled />
    </div>
  )
}

export default App
```

error TS2307: Cannot find module '@codesandbox/sandpack-react/unstyled' or its corresponding type declarations.

--- a/sandpack-react/package.json
+++ b/sandpack-react/package.json
@@ -10,11 +10,11 @@
   "license": "Apache-2.0",
   "author": "CodeSandbox",
   "main": "dist/index.js",
-  "module": "dist/index.mjs",
+  "module": "dist/index.esm.js",
   "types": "dist/index.d.ts",
   "exports": {
     "./unstyled": {
-      "types": "./dist/unstyled/index.d.ts",
+      "types": "./dist/unstyled.d.ts",
       "import": "./dist/unstyled/index.esm.js",
       "require": "./dist/unstyled/index.js"
     },
@@ -22,6 +22,14 @@
       "types": "./dist/index.d.ts",
       "import": "./dist/index.esm.js",
       "require": "./dist/index.js"
+    }
+  },
+  "typesVersions": {
+    "*": {
+      "*": [
+        "./dist/*",
+        "./*"
+      ]
     }
   },
   "scripts": {

--- a/sandpack-react/package.json
+++ b/sandpack-react/package.json
@@ -14,7 +14,7 @@
   "types": "dist/index.d.ts",
   "exports": {
     "./unstyled": {
-      "types": "./dist/unstyled.d.ts",
+      "types": "./dist/unstyled/index.d.ts",
       "import": "./dist/unstyled/index.esm.js",
       "require": "./dist/unstyled/index.js"
     },

--- a/sandpack-react/package.json
+++ b/sandpack-react/package.json
@@ -14,7 +14,7 @@
   "types": "dist/index.d.ts",
   "exports": {
     "./unstyled": {
-      "types": "./dist/unstyled/index.d.ts",
+      "types": "./dist/unstyled.d.ts",
       "import": "./dist/unstyled/index.esm.js",
       "require": "./dist/unstyled/index.js"
     },

--- a/sandpack-react/rollup.config.js
+++ b/sandpack-react/rollup.config.js
@@ -5,6 +5,7 @@ const typescript = require("@rollup/plugin-typescript");
 const filesize = require("rollup-plugin-filesize");
 
 const pkg = require("./package.json");
+const generateUnstyledTypes = require("./scripts/rollup-generate-unstyled-types");
 const removeCss = require("./scripts/rollup-remove-css-transformer");
 
 const basePlugins = [commonjs({ requireReturnsDefault: "preferred" })];
@@ -70,6 +71,7 @@ const configBase = [
         compilerOptions: { outDir: "dist/unstyled/" },
       }),
       removeCss(),
+      generateUnstyledTypes(),
       filesize()
     ),
     output: [

--- a/sandpack-react/scripts/rollup-generate-unstyled-types.js
+++ b/sandpack-react/scripts/rollup-generate-unstyled-types.js
@@ -1,0 +1,14 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const fs = require("fs");
+
+module.exports = function () {
+  return {
+    name: "remove-css",
+    buildEnd() {
+      fs.writeFileSync(
+        "dist/unstyled.d.ts",
+        `export * from './unstyled/index';`
+      );
+    },
+  };
+};

